src/main/java/net/imglib2/img/array/ArrayImgs.java

```diff
@@ -353,6 +370,111 @@ private ArrayImgs()
+	{
+		final ArrayImg< Unsigned4BitType, A > img = new ArrayImg<>( access, dim, new Fraction( 1, 64 ) );
+		final Unsigned4BitType t = new Unsigned4BitType( img );
+		img.setLinkedType( t );
+		return img;
+	}
+
+	/**
+	 * Create an {@link ArrayImg}&lt;{@link Unsigned12BitType}, {@link LongArray}&gt;.
+	 */
+	@SuppressWarnings( "unchecked" )
+	final static public ArrayImg< Unsigned12BitType, LongArray > unsigned12Bits( final long... dim )
+	{
+		return ( ArrayImg< Unsigned12BitType, LongArray > ) new ArrayImgFactory< Unsigned12BitType >().create( dim, new Unsigned12BitType() );
+	}
+
+	/**
+	 * Creates an {@link ArrayImg}&lt;{@link Unsigned12BitType}, {@link LongAccess}&gt;
+	 * using a {@link LongAccess} passed as argument.
+	 */
+	final static public < A extends LongAccess > ArrayImg< Unsigned12BitType, A > unsigned12Bits( final A access, final long... dim )
+	{
+		final ArrayImg< Unsigned12BitType, A > img = new ArrayImg<>( access, dim, new Fraction( 1, 64 ) );
+		final Unsigned12BitType t = new Unsigned12BitType( img );
+		img.setLinkedType( t );
+		return img;
+	}
+
+	/**
+	 * Create an {@link ArrayImg}&lt;{@link Unsigned128BitType}, {@link LongArray}&gt;.
+	 */
+	@SuppressWarnings( "unchecked" )
+	final static public ArrayImg< Unsigned128BitType, LongArray > unsigned128Bits( final long... dim )
+	{
+		return ( ArrayImg< Unsigned128BitType, LongArray > ) new ArrayImgFactory< Unsigned128BitType >().create( dim, new Unsigned128BitType() );
+	}
+
+	/**
+	 * Creates an {@link ArrayImg}&lt;{@link Unsigned128BitType}, {@link LongAccess}&gt;
+	 * using a {@link LongAccess} passed as argument.
+	 */
+	final static public < A extends LongAccess > ArrayImg< Unsigned128BitType, A > unsigned128Bits( final A access, final long... dim )
+	{
+		final ArrayImg< Unsigned128BitType, A > img = new ArrayImg<>( access, dim, new Fraction( 1, 64 ) );
+		final Unsigned128BitType t = new Unsigned128BitType( img );
+		img.setLinkedType( t );
+		return img;
+	}
+
+	/**
+	 * Create an {@link ArrayImg}&lt;{@link UnsignedVariableBitLengthType}, {@link LongArray}&gt;.
+	 */
+	@SuppressWarnings( "unchecked" )
+	final static public ArrayImg< UnsignedVariableBitLengthType, LongArray > unsignedVariableBitLengths(final int nbits, final long... dim )
+	{
+		return ( ArrayImg< UnsignedVariableBitLengthType, LongArray > ) new ArrayImgFactory< UnsignedVariableBitLengthType >().create( dim, new UnsignedVariableBitLengthType(nbits) );
+	}
+
+	/**
+	 * Creates an {@link ArrayImg}&lt;{@link UnsignedVariableBitLengthType}, {@link LongAccess}&gt;
+	 * using a {@link LongAccess} passed as argument.
+	 */
+	final static public < A extends LongAccess > ArrayImg< UnsignedVariableBitLengthType, A > unsignedVariableBitLengths( final A access, final int nbits, final long... dim )
+	{
+		final ArrayImg< UnsignedVariableBitLengthType, A > img = new ArrayImg<>( access, dim, new Fraction( 1, 64 ) );
```

Comment:
Shouldn't this be `new Fraction( t.getBitsPerPixel(), 64 )` instead of `new Fraction( 1, 64 )` here?